clients/geth/specular/cmd/utils/flags.go

```diff
@@ -129,39 +165,63 @@ func RegisterEthService(stack *node.Node, cfg *ethconfig.Config) (ethapi.Backend
+		validatorAddr       common.Address
+		sequencerPassphrase string
+		validatorPassphrase string
+	)
+	pwList := utils.MakePasswordList(ctx)
+	clefEndpoint := ctx.String(RollupL2ClefEndpointFlag.Name)
+	if clefEndpoint == "" {
+		if len(pwList) == 0 {
+			utils.Fatalf("Failed to register rollup services: no clef endpoint or password provided")
+		}
+		if ctx.String(RollupSequencerAddrFlag.Name) != "" {
+			sequencerAddr = common.HexToAddress(ctx.String(RollupSequencerAddrFlag.Name))
+			sequencerPassphrase = pwList[0]
+			pwList = pwList[1:]
+		}
+		if ctx.String(RollupValidatorAddrFlag.Name) != "" {
+			validatorAddr = common.HexToAddress(ctx.String(RollupValidatorAddrFlag.Name))
+			validatorPassphrase = pwList[0]
+			pwList = pwList[1:]
+		}
 	}
 
-	node := ctx.String(RollupNodeFlag.Name)
-	coinbase := common.HexToAddress(ctx.String(RollupCoinBaseFlag.Name))
-	sequencerAddr := common.HexToAddress(ctx.String(RollupSequencerAddrFlag.Name))
-	if node == "sequencer" && sequencerAddr == (common.Address{}) {
-		sequencerAddr = coinbase
-	}
-	if sequencerAddr == (common.Address{}) {
-		utils.Fatalf("Failed to register the Rollup service: sequencer address not specified")
-	}
-	cfg := &rollup.Config{
-		Node:                 node,
-		Coinbase:             coinbase,
-		Passphrase:           passphrase,
-		ClefEndpoint:         ctx.String(RollupClefEndpointFlag.Name),
-		L1Endpoint:           ctx.String(RollupL1EndpointFlag.Name),
-		L1ChainID:            ctx.Uint64(RollupL1ChainIDFlag.Name),
-		SequencerAddr:        sequencerAddr,
-		SequencerInboxAddr:   common.HexToAddress(ctx.String(RollupSequencerInboxAddrFlag.Name)),
-		RollupAddr:           common.HexToAddress(ctx.String(RollupRollupAddrFlag.Name)),
-		L1RollupGenesisBlock: ctx.Uint64(RollupL1RollupGenesisBlock.Name),
-		RollupStakeAmount:    ctx.Uint64(RollupRollupStakeAmount.Name),
-	}
-	return cfg
+	return &rollup.SystemConfig{
+		L1Config: rollup.L1Config{
+			L1Endpoint:           ctx.String(RollupL1EndpointFlag.Name),
+			L1ChainID:            ctx.Uint64(RollupL1ChainIDFlag.Name),
+			L1RollupGenesisBlock: ctx.Uint64(RollupL1RollupGenesisBlockFlag.Name),
+			SequencerInboxAddr:   common.HexToAddress(ctx.String(RollupSequencerInboxAddrFlag.Name)),
+			RollupAddr:           common.HexToAddress(ctx.String(RollupRollupAddrFlag.Name)),
+		},
+		L2Config: rollup.L2Config{
+			L2Endpoint:     "ws://0.0.0.0:4012", // TODO: read this from http params? or from a separate flag
+			L2ClefEndpoint: clefEndpoint,
+		},
+		SequencerConfig: rollup.SequencerConfig{
+			SequencerAccountAddr: sequencerAddr,
+			SequencerPassphrase:  sequencerPassphrase,
+			MinExecutionInterval: time.Duration(ctx.Uint64(RollupSequencerMinExecutionIntervalFlag.Name)) * time.Second,
+			MaxExecutionInterval: time.Duration(ctx.Uint64(RollupSequencerMaxExecutionIntervalFlag.Name)) * time.Second,
+			SequencingInterval:   time.Duration(ctx.Uint64(RollupSequencerSequencingIntervalFlag.Name)) * time.Second,
```

Comment:
I ran into this while working with the refactored code. It seems like this ends up not using the default value.
`time.Duration(RollupSequencerSequencingIntervalFlag.Value) * time.Second` works for me.

Reply:
oh right good catch, i'll move the defaults into the flags